### ISSUE TITLE
current rng setup is full of footguns in jax

### ISSUE THREAD

**Issue by GallagherCommaJack** [keras-team/keras#18426]
right now unseeded calls to e.g. `keras.random.uniform` are going to acquire static seeds at trace time. this has a few undesirable consequences:

1) subsequent calls will have the same randomness each time (e.g. dropout will have a fixed mask instead of random each step)
2) the jax compiler cache will ~never hit, as the constant rng seed values will be different every time

to get around this, some kind of rng state management is necessary. flax does this with hierarchical management of rng's from the `Scope`. such an approach is fairly complex however, and there might be simpler options e.g. a single global `rng` state, which gets included with the training state in `model.fit`, unseeded rng calls would then do something along the lines of

```
state.seed, local_seed = jax.random.split(state.seed)
```

**Comment by aaarrti** [comment]
Hi @GallagherCommaJack @fchollet, 
do you plan to resume work on this issue? What is the recommended way of handling RNGs? 
I believe it would be much appreciated, if you could add a documentation (or tutorial) covering this topic 🙃.

**Comment by rivershah** [comment]
Related so putting this here rather than new issue. `SeedGenerator` is returning same value per-replica. This is not good behaviour when training on larger tpu clusters.

```
import tensorflow as tf
from keras import random

strategy = ...TPUStrategy instantiation code
with strategy.scope():
    seed = random.SeedGenerator()


@tf.function(jit_compile=True)
def get_seed():
    return seed.next()


output = strategy.run(get_seed)
print(output) # This will be same for each replica which is not desired
```

**Comment by pctablet505** [comment]
Unseeded random ops in JAX tracing now raise a clear ValueError. keras.src.random.seed_generator.global_seed_generator lines 122-137 raise: '[JAX RNG] When tracing a JAX function, you should only use seeded random ops...' Docstring of keras.random.uniform also documents this. The global-seed footgun is blocked.
